src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,107 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceImplTest {
+    private RegistrationService registrationService;
+    private StorageDao storageDao;
+
+    @BeforeEach
+    public void setUp() {
+        storageDao = new StorageDaoImpl();
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @Test
+    public void register_validUser_ok() {
+        // Create a user with unique login for the test
+        User user = new User();
+        user.setLogin("john_doe_" + System.currentTimeMillis());
+        user.setPassword("strong_password");
+        user.setAge(25);
+        User registeredUser = registrationService.register(user);
+        assertNotNull(registeredUser.getId());
+        assertEquals(user.getLogin(), registeredUser.getLogin());
+        assertEquals(user.getPassword(), registeredUser.getPassword());
+        assertEquals(user.getAge(), registeredUser.getAge());
+        User retrievedUser = storageDao.get(user.getLogin());
+        assertNotNull(retrievedUser);
+        assertEquals(registeredUser.getId(), retrievedUser.getId());
+        assertEquals(registeredUser.getLogin(), retrievedUser.getLogin());
+        assertEquals(registeredUser.getPassword(), retrievedUser.getPassword());
+        assertEquals(registeredUser.getAge(), retrievedUser.getAge());
+    }
+
+    @Test
+    public void register_nullLogin_notOk() {
+        User user = new User();
+        user.setPassword("strong_password");
+        user.setAge(25);
```

Comment:
to avoid code duplication you can create valid user BeforeEach and set one invalid field in each test to him